installer/Util/SystemConfig.php

```diff
@@ -663,4 +677,90 @@ private function isExcluded(string $currentVersion, array $excludedRange = []):
         }
         return false;
     }
+
+    /**
+     * Return PHP version
+     * @return string
+     */
+    public function getPhpVersion(): string
+    {
+        return phpversion();
+    }
+
+    /**
+     * Return web server details
+     * @return string
+     */
+    public function getWebServerDetails(): string
+    {
+        return $_SERVER['SERVER_SOFTWARE'];
+    }
+
+    /**
+     * Return MySql client version
+     * @return string
+     */
+    public function getMysqlClientVersion(): string
+    {
+        $mysqlClient = mysqli_get_client_info();
+        $versionPattern = '/[0-9]+\.[0-9]+\.[0-9]+/';
+        preg_match($versionPattern, $mysqlClient, $matches);
+        return $matches[0];
+    }
+
+    /**
+     * Return MySql server version
+     * @return string
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function getMysqlServerVersion(): string
+    {
+        $connection = $this->upgraderConfigUtility->getConnection();
+        $result = $connection->executeQuery("SELECT VERSION() AS version")->fetchAssociative();
+        return $result['version'];
+    }
+
+    /**
+     * @return string
+     */
+    public function getMySqlHostInfo(): string
+    {
+        return 'Localhost via UNIX socket';
```

Comment:
Just hard coded since I still finding a way to get host info with DBAL.
In mysql there we can get this using "$mysqli->host_info",
ref: https://www.php.net/manual/en/mysqli.get-host-info.php
In PDO there we can get this using "PDO::getAttribute(PDO::ATTR_CONNECTION_STATUS)"
ref: https://www.php.net/manual/en/pdo.getattribute.php